craft_application/application.py

```diff
@@ -601,6 +608,14 @@ def _pre_run(self, dispatcher: craft_cli.Dispatcher) -> None:
             self._enable_fetch_service = True
             self._fetch_service_policy = fetch_service_policy
 
+        command = cast(
+            commands.AppCommand,
+            dispatcher.load_command(self.app_config),
+        )
+
+        if command.name == "test":
```

Comment:
I don't think this approach is sustainable. If the application has to be aware of specific commands we're in big trouble

Reply:
Agreed. As discussed with @lengau we're adding hacks to work around limitations in the current architecture. We'll leave this as is for now and address it when refactoring the infrastructure.